src/net/ossl.cc

```diff
@@ -815,11 +818,17 @@ class session : public enable_shared_from_this<session>, public session_impl {
         assert(_output_pending.available());
     }
 
+    const char * get_type_string() const {
+        return _type == session_type::CLIENT ? "Client": "Server";
+    }
+
     // This function waits for the _output_pending future to resolve
     // If an error occurs, it is saved off into _error and returned
     future<> wait_for_output() {
+        _logger.trace("wait_for_output");
```

Comment:
Why invent a new logger type when you could just add a formatter for the session object that pretty-prints the local/remote etc, and then just use `tls_log::trace("{} wait_for_output")`?

Reply:
Thanks for the suggestion, great idea!